packages/spacecat-shared-slack-client/src/index.js

```diff
@@ -9,77 +9,64 @@
  * OF ANY KIND, either express or implied. See the License for the specific language
  * governing permissions and limitations under the License.
  */
-import { WebClient } from '@slack/web-api';
 import { hasText, isObject } from '@adobe/spacecat-shared-utils';
 
+import BaseSlackClient from './clients/base-slack-client.js';
+import ElevatedSlackClient from './clients/elevated-slack-client.js';
+
 const ENV_PREFIX = 'SLACK_TOKEN_';
 
 export const SLACK_TARGETS = {
   ADOBE_INTERNAL: 'ADOBE_INTERNAL',
   ADOBE_EXTERNAL: 'ADOBE_EXTERNAL',
 };
 
-function getEnvironmentVariableNameForTarget(target) {
-  return `${ENV_PREFIX}${target}`;
+function getEnvironmentVariableNameForTarget(target, isElevated = false) {
+  return `${ENV_PREFIX}${target}${isElevated ? '_ELEVATED' : ''}`;
 }
 
-export class SlackClient {
-  static createFrom(context, target) {
-    const { log } = context;
+function getOpsConfig(context, target) {
+  const opsChannelId = context.env[`SLACK_OPS_CHANNEL_${target}`];
+  const admins = (context.env[`SLACK_OPS_ADMINS_${target}`] || '')
+    .split(',')
+    .filter((admin) => hasText(admin));
 
-    if (!hasText(target)) {
-      throw new Error('Missing target for the Slack Client');
-    }
+  if (!hasText(opsChannelId)) {
+    throw new Error(`No Ops Channel ID set for ${target}`);
+  }
 
-    if (!isObject(context.slackClients)) {
-      context.slackClients = {};
-    }
+  return { opsChannelId, admins };
+}
 
-    if (context.slackClients[target]) {
-      return context.slackClients[target];
-    }
+function getToken(context, target, isElevated) {
+  const token = context.env[getEnvironmentVariableNameForTarget(target, isElevated)];
+  if (!hasText(token)) {
+    throw new Error(`No Slack token set for ${target}${isElevated ? ' with elevated privileges' : ''}`);
+  }
+  return token;
+}
 
-    const token = context.env[getEnvironmentVariableNameForTarget(target)];
-    if (!hasText(token)) {
-      throw new Error(`No slack token set for ${target}`);
-    }
+const createFrom = (context, target, isElevated = false) => {
```

Comment:
return type of the factory method changes based on the value of `isElevated` argument. We should move `createFrom` factory method to respective classes. If the consumer wants to create an `ElevatedSlackClient`, they should declare it explicitly:

```
const slackClient = ElevatedSlackClient.createFrom(context, target);
```

instead of more implicit:

```
const slackClient = createFrom(context, target, true);
```